src/reformed.js

```diff
@@ -45,11 +46,26 @@ const makeWrapper = (middleware) => (WrappedComponent) => {
       }
     }
 
-    bindInput = (name) => {
+    bindInput = (name, opts) => {
+      const cfg = assign({
+        updateOn: 'onChange',
+        debounce: 0
+      }, opts)
+
+      const { updateOn } = cfg
+      const asyncHandler = (fn) => (e) => {
+        e.persist()
+        fn(e)
+      }
+
+      const updater = (!cfg.debounce)
+        ? this.bindToChangeEvent
+        : asyncHandler(debounce(this.bindToChangeEvent, cfg.debounce, false))
+
       return {
         name,
-        value: this.state.model[name] || '',
-        onChange: this.bindToChangeEvent,
+        defaultValue: this.state.model[name] || '',
```

Comment:
Yeah, I was looking at a few options last night. Perhaps a better approach would be to include the `updateOn` and `debounce` options in the validation config (called `validationSchema` in the demo). But then we would need the `reformed` component to expose more data to the  `validateSchema` component for it to make determinations about form interactions (similar to Angular's pristine/dirty, touched/untouched). I'm "thinking out loud" here mostly.